xoto3/dynamodb/write_versioned/api2.py

```diff
@@ -0,0 +1,132 @@
+"""
+A table and type-oriented API for using versioned_transact_write_items.
+
+All of these things are simple bolt-ons for the underlying
+functionality, and may safely be ignored.
+"""
+from copy import deepcopy
+from typing import Callable, Generic, Optional, TypeVar, Union
+
+from .modify import define_table, delete, presume, put
+from .read import get, require
+from .types import Item, ItemKey, TransactionBuilder, VersionedTransaction
+
+T = TypeVar("T")
+
+
+class TypedTable(Generic[T]):
+    """In practice, it's nice to have table-aware closures of these
+    functions when dealing with a table, versus providing the string
+    table name to every call. This is especially valuable if you're
+    dealing with only a single table over and over again, and there's
+    some kind of ser/de you want to run consistently against that
```

Comment:
`ser/de`?

Reply:
Is it `serialize/deserialize`?

Reply:
yep. i can spell this out if that would be better. `serde` is a fairly common abbreviation in my world, but that doesn't mean it's the right approach for documentation.